File: app/src/main/java/cc/smartcash/wallet/Adapters/ContactAdapter.java
package cc.smartcash.wallet.Adapters;

import android.content.Context;
import android.database.Cursor;
import android.net.Uri;
import android.provider.ContactsContract;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.CursorAdapter;
import android.widget.QuickContactBadge;
import android.widget.TextView;

import com.squareup.picasso.Picasso;

import cc.smartcash.wallet.R;

public class ContactAdapter extends CursorAdapter {

    int[] indices;

    public ContactAdapter(Context context, Cursor c) {
        super(context, c);
    }

    @Override
    public View newView(Context context, Cursor cursor, ViewGroup parent) {
        indices = new int[]{
                cursor.getColumnIndex(ContactsContract.Contacts._ID),
                cursor.getColumnIndex(ContactsContract.Contacts.LOOKUP_KEY),
                cursor.getColumnIndex(ContactsContract.Contacts.DISPLAY_NAME)
        };
        return LayoutInflater.from(context).inflate(R.layout.contact_dialog_item, null);
    }

    @Override
    public void bindView(View view, Context context, Cursor cursor) {
        TextView txtNome = view.findViewById(R.id.txtContactName);
        QuickContactBadge quickContactBadge = view.findViewById(R.id.qcbFoto);
        Uri uriContact = ContactsContract.Contacts.getLookupUri(cursor.getLong(indices[0]), cursor.getString(indices[1]));
        txtNome.setText(cursor.getString(indices[2]));
        quickContactBadge.assignContactUri(uriContact);
        Picasso.get().load(uriContact).placeholder(R.mipmap.ic_launcher).into(quickContactBadge);
    }
}

File: app/src/main/java/cc/smartcash/smarthub/adapters/ContactAdapter.kt
package cc.smartcash.smarthub.adapters

import android.content.Context
import android.database.Cursor
import android.provider.ContactsContract
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.CursorAdapter
import android.widget.QuickContactBadge
import android.widget.TextView
import cc.smartcash.smarthub.R
import com.squareup.picasso.Picasso

class ContactAdapter(context: Context, c: Cursor) : CursorAdapter(context, c) {

    private var indices: IntArray? = null

    override fun newView(context: Context, cursor: Cursor, parent: ViewGroup): View {
        indices = intArrayOf(cursor.getColumnIndex(ContactsContract.Contacts._ID), cursor.getColumnIndex(ContactsContract.Contacts.LOOKUP_KEY), cursor.getColumnIndex(ContactsContract.Contacts.DISPLAY_NAME))
        return LayoutInflater.from(context).inflate(R.layout.contact_dialog_item, null)
    }

    override fun bindView(view: View, context: Context, cursor: Cursor) {
        val txtNome = view.findViewById<TextView>(R.id.txtContactName)
        val quickContactBadge = view.findViewById<QuickContactBadge>(R.id.qcbPhoto)
        val uriContact = ContactsContract.Contacts.getLookupUri(cursor.getLong(indices!![0]), cursor.getString(indices!![1]))
        txtNome.text = cursor.getString(indices!![2])
        quickContactBadge.assignContactUri(uriContact)
        Picasso.get().load(uriContact).placeholder(R.mipmap.ic_launcher).into(quickContactBadge)
    }
}
